### ISSUE TITLE
decorate_events is missing from v3 and v4

### ISSUE THREAD

**Issue by talevy** [logstash-plugins/logstash-input-kafka#122]
The V3 and V4 version branches of this plugin are lacking the `decorate_events` configuration option:

```
% git tag | xargs -n1 sh -c 'git checkout $1 > /dev/null 2>&1; if ack 'decorate_events' lib > /dev/null; then echo -n YES:; else echo -n NO:; fi; echo $1' -

YES:v2.0.4
YES:v2.0.6
YES:v2.0.7
YES:v2.0.8
YES:v2.0.9
YES:v3.0.0
NO:v3.0.0.beta5
NO:v3.0.0.beta6
NO:v3.0.0.beta7
YES:v3.0.1
NO:v3.0.2
NO:v4.0.0
NO:v5.0.1
YES:v5.0.2
YES:v5.0.3
```

bash cred: @jordansissel 

need to add this into `v4.1` and `v3.1`.

**Comment by pbkdf3** [comment]
Also missing topics_pattern regex based subscriptions, which are necessary in many deployments.
Both could be ported from recent commits to 5.X.
I did so, but am waiting for my company to decide its OK to sign the Elastic CLA before submitting (and who knows when that will happen).  
But it was easy and I'm sure anyone else could do the same.

**Comment by pbkdf3** [comment]
v4 still needs both of these.  the rest seem to have it.  But for anyone who needs to use kafka .9 and wants to use logstash 5 v4 is the only option.  Please port the features decorate_events and regex topic subscriptions to v4.  Thanks you!

**Comment by lovromazgon** [comment]
Is this feature coming to v4.1 soon? We would need it because we are using Kafka 0.9. Thanks.

**Comment by pbkdf3** [comment]
Yes! please see this, it covers both decorate_events and topics_pattern:

https://github.com/logstash-plugins/logstash-input-kafka/issues/173

Currently Kafka .9 users (like myself and OP) are stuck with logstash 2.4
in order to have reasonable Kafka feature coverage.

**Comment by suyograo** [comment]
This has been backported to 4.x and released as 4.2.0. I am not sure it's worth backporting to 3.x;
